src/Gadgets.ts

```diff
@@ -0,0 +1,90 @@
+import Downloader from './Downloader.js'
+import * as logger from './Logger.js'
+import MediaWiki from './MediaWiki.js'
+
+class Gadgets {
+  private static instance: Gadgets
+  private gadgets: Gadget[]
+
+  public static getInstance() {
+    if (!Gadgets.instance) {
+      Gadgets.instance = new Gadgets()
+    }
+    return Gadgets.instance
+  }
+
+  public async fetchGadgets() {
+    const gadgetsUrl = Downloader.apiUrlDirector.buildGadgetsUrl()
+    const gadgetsResponse = await Downloader.getJSON<GadgetQueryResult>(gadgetsUrl)
+    if (!gadgetsResponse.batchcomplete) {
+      throw new Error(`Error while fetching gadgets with ${gadgetsUrl}, batch seems to be incomplete, scraper does not expect/support pagination here`)
+    }
+    this.gadgets = gadgetsResponse.query.gadgets
+    logger.info(`${this.gadgets.length} gadgets retrieved`)
+  }
+
+  public getCssOnlyGadgets(articleDetail: ArticleDetail) {
```

Comment:
This doesn't account for `peer` dependencies as described in https://github.com/openzim/mwoffliner/issues/2212#issuecomment-2978175587 yet.

Reply:
Actually, as long as we don't load JS due to #2310, we can just ignore the peers as well. Only support for JS modules is added, we will need to support peers as well

Reply:
Let's add them, I don't think it will do any harm, and I'm pretty sure we have high chances to forget the peers once we add support fo JS modules.

Reply:
This is what I did indeed.